wdl-analysis/src/types/v1.rs

```diff
@@ -1478,20 +1522,22 @@ impl<'a, C: EvaluationContext> ExprTypeEvaluator<'a, C> {
                                 target.text(),
                                 target.span(),
                                 maximum,
-                                arguments.len(),
+                                count,
                                 expr.arguments().skip(maximum).map(|e| e.span()),
                             ));
                         }
                         Err(FunctionBindError::ArgumentTypeMismatch { index, expected }) => {
-                            self.context.add_diagnostic(argument_type_mismatch(
-                                target.text(),
-                                &expected,
-                                &arguments[index],
-                                expr.arguments()
-                                    .nth(index)
-                                    .map(|e| e.span())
-                                    .expect("should have span"),
-                            ));
+                            if index < arguments.len() {
```

Comment:
It should be guaranteed that `index < arguments.len()`, so I would rather panic when that's not the case than silently not emit a diagnostic.

I think we can revert this check.